src/WebJobs.Script/Host/BlobLeaseManager.cs

```diff
@@ -26,22 +26,25 @@ internal sealed class BlobLeaseManager : IDisposable
         private readonly ILogger _logger;
         private readonly string _hostId;
         private readonly string _instanceId;
-        private ICloudBlob _lockBlob;
-        private string _leaseId;
+        private IDistributedLock _lockHandle; // If non-null, then we own the lock.
+
         private bool _disposed;
         private bool _processingLease;
         private DateTime _lastRenewal;
         private TimeSpan _lastRenewalLatency;
 
-        internal BlobLeaseManager(ICloudBlob lockBlob, TimeSpan leaseTimeout, string hostId, string instanceId, TraceWriter traceWriter,
+        private IDistributedLockManager _lockManager;
+
+        internal BlobLeaseManager(IDistributedLockManager lockManager, TimeSpan leaseTimeout, string hostId, string instanceId, TraceWriter traceWriter,
```

Comment:
Will rename to "PrimaryHostCoordinator" 

---
In reply to: [123669617](https://github.com/Azure/azure-webjobs-sdk-script/pull/1615#discussion_r123669617) [](ancestors = 123669617)